### PR TITLE
Use MetadataSegmentManager info during segment move.

### PR DESCRIPTION
**Pull request by akashdw**
PR to get segment information from `MetadataSegmentManager`  instead of getting it from`fromServer's`. This is useful when `MetadataSegmentManager` and `fromServer` DataSegment's are different for same identifier  (say loadSpec differs because of deep storage migration).

### CODE REVIEW

**Comment by akashdw** [comment]
test timed out, reopening PR.

File: server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java [hunk 1]
```diff
@@ -362,22 +362,30 @@ public String getCurrentLeader()
...
 
-      final DataSegment segment = fromServer.getSegment(segmentName);
-      if (segment == null) {
-        throw new IAE("Unable to find segment [%s] on server [%s]", segmentName, fromServer.getName());
+      DruidDataSource dataSource = metadataSegmentManager.getInventoryValue(segment.getDataSource());
+      if (dataSource == null) {
+        throw new IAE("Unable to find dataSource for segment [%s] in metadata", segmentName);
       }
 
+      final DataSegment segmentToLoad = dataSource.getSegment(segment.getIdentifier());
```
Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java]: segmentToLoad == null case?

Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java]: also, can you add some comment as to why DataSegment is re-obtained even if we got it in the method arg?

Comment (himanshug) [inline server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java]: that is, explaining the case when they would be different.

**Comment by akashdw** [comment]
@himanshug Addressed comments.

**Comment by himanshug** [comment]
:+1:

**Pull request merged**